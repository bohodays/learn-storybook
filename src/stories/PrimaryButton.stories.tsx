import type { Meta, StoryObj } from "@storybook/react-vite";

import { fn } from "storybook/test";
import PrimaryButton from "../components/PrimaryButton";

// Docs 정보
const meta = {
  title: "Buttons/PrimaryButton", // 경로를 나타냄
  component: PrimaryButton,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  decorators: [
    (Story) => (
      <div style={{ width: "360px" }}>
        <Story />
      </div>
    ),
  ],
  argTypes: {
    theme: {
      control: {
        type: "select",
        options: ["dark", "light", "social", "text"],
      },
      description: "버튼 테마",
      defaultValue: "dark",
    },
    isDisabled: {
      control: "boolean",
      description: "버튼 비활성화 여부",
      defaultValue: false,
    },
    children: { control: "text", description: "버튼 text" },
    onClick: { action: "clicked", description: "버튼 클릭 이벤트" },
  },
  args: { onClick: fn() },
} satisfies Meta<typeof PrimaryButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Dark: Story = {
  args: {
    children: "Button",
    theme: "dark",
    isDisabled: false,
  },
};

export const Light: Story = {
  args: {
    children: "Button",
    theme: "light",
    isDisabled: false,
  },
};

export const Social: Story = {
  args: {
    children: "Button",
    theme: "social",
    isDisabled: false,
  },
};

export const Text: Story = {
  args: {
    children: "Button",
    theme: "text",
    isDisabled: false,
  },
};

export const Disabled: Story = {
  args: {
    children: "Button",
    theme: "dark",
    isDisabled: true,
  },
};
