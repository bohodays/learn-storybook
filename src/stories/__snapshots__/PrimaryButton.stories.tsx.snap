// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/PrimaryButton Dark smoke-test 1`] = `
<div style="width: 360px;">
  <button class="w-full  h-[59px] rounded-primary-button bg-primary text-white disabled:bg-mono100 disabled:text-mono200">
    Button
  </button>
</div>
`;

exports[`Buttons/PrimaryButton Disabled smoke-test 1`] = `
<div style="width: 360px;">
  <button class="w-full  h-[59px] rounded-primary-button bg-primary text-white disabled:bg-mono100 disabled:text-mono200"
          disabled
  >
    Button
  </button>
</div>
`;

exports[`Buttons/PrimaryButton Light smoke-test 1`] = `
<div style="width: 360px;">
  <button class="w-full  h-[59px] rounded-primary-button bg-white text-primary disabled:bg-mono100 disabled:text-mono200">
    Button
  </button>
</div>
`;

exports[`Buttons/PrimaryButton Social smoke-test 1`] = `
<div style="width: 360px;">
  <button class="w-full  h-[59px] rounded-primary-button bg-social text-white disabled:bg-mono100 disabled:text-mono200">
    Button
  </button>
</div>
`;

exports[`Buttons/PrimaryButton Text smoke-test 1`] = `
<div style="width: 360px;">
  <button class="w-full  h-[59px] rounded-primary-button bg-transparent text-white disabled:bg-mono100 disabled:text-mono200">
    Button
  </button>
</div>
`;
